transport/http/inbound.go

```diff
@@ -55,9 +55,10 @@ func Mux(pattern string, mux *http.ServeMux) InboundOption {
 // sharing this transport.
 func (t *Transport) NewInbound(addr string, opts ...InboundOption) *Inbound {
 	i := &Inbound{
-		once:   sync.Once(),
-		addr:   addr,
-		tracer: t.tracer,
+		once:      sync.Once(),
+		addr:      addr,
+		tracer:    t.tracer,
+		transport: t,
```

Comment:
Unkeyed fields really hurt readability with large structs. I would much rather have the readability.

If we're concerned about missing fields, it's not unfeasible to create a linter for that.